### DISCUSSION TITLE
About InitAddress, how to know which address was reached?

### DISCUSSION THREAD

**Discussion by juancresc** [redis/rueidis#205] [General]
I'm using several servers to read from:

```
var client, err = rueidis.NewClient(rueidis.ClientOption{
	InitAddress: []string{
```

And whenever I use the get command

`res, err := client.Do(ctx, client.B().Get().Key("1234").Build()).ToString()`

I'm trying to figure out which of the InitAddress was used (to have metrics). My idea is to use this to replace the Ring client from go-redis. 
What is a good way to do it? 
How is each address get picked?
What happens if an address is down, is it removed?

**Comment by rueian** [comment]
Hi @juancresc,

rueidis follows the Redis cluster spec https://redis.io/docs/reference/cluster-spec/ and always reads from primary Redis nodes, not replicas. It is not compatible with the Ring implementation of go-redis. I would strongly recommend you to use a standard Redis cluster instead.

Following the spec, rueidis will pick a node who is responsible for the requested key slot. If the node is down or removed, rueidis will find its successor through cluster protocol.

**Reply by juancresc** [reply]
Hi!

when you say a standard Redis cluster, you mean a rueidis cluster client or another library?

**Reply by rueian** [reply]
I mean this https://redis.io/docs/management/scaling/

**Reply by juancresc** [reply]
I see. For our specific use case we use kvrocks with a master/slave arch. That's why we use a Ring client so far. You think rueidis is not suitable for this? Thanks again!

**Reply by rueian** [reply]
No, rueidis is not suitable for replacing the Ring implementation of go-redis.

On the other hand, Kvrocks supports Redis cluster spec: https://kvrocks.apache.org/docs/cluster/. I believe it will be better than client side Ring.
